app/controllers/admin/orders_controller.rb

```diff
@@ -1,10 +1,14 @@
 class Admin::OrdersController < Admin::BaseController
-  before_action :load_order, only: %i(update show check_valid_status?)
-  before_action :check_valid_status?, :get_status_name, only: :update
+  before_action :load_order, except: %i(index)
+  before_action :check_valid_status_for_update?, :get_status_name, only: :update
+  before_action :check_valid_status_for_destroy?, only: :destroy
 
   def index
     @pagy, @orders = pagy Order.recent_orders,
                           items: Settings.length.per_page_5
+    @count_orders_by_status = count_orders_by_status
+    @pagy, @orders = pagy @orders.filter_by_name(username) if username.present?
+    @pagy, @orders = pagy Order.send(status) if status.present?
```

Comment:
ủa sao nó lặp lặp vậy cu ?

```suggestion
    @pagy, @orders =  if username.present?
        pagy @orders.filter_by_name(username)
    elsif status.present?
       pagy Order.send(status)
    end
```